server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -203,6 +203,8 @@ public static HighlightBuilder highlight() {
 
     private Map<String, Object> runtimeMappings = emptyMap();
 
+    private boolean hideSourceFields = true;
```

Comment:
Can we change the name of the flag to be more descriptive? Like `hideInferenceResultFields`?